### PR TITLE
Fix gas price

### PR DESCRIPTION
**Pull request by apoorvsadana**

### CODE REVIEW

File: crates/node/src/genesis_block.rs [hunk 1]
```diff
@@ -60,8 +60,10 @@ fn construct_genesis_block<Block: BlockT>(state_root: Block::Hash, state_version
 
     let mut digest = vec![];
     let block = StarknetBlock::try_new(
-        // TODO: Decide what values to put here
-        // This is just filler values for now
+        // The genesis block values don't really matter because when you start a node,
+        // you need to disable fees completely to do the setup process. So we need some blocks
+        // of txs without any fees. And the 1st block will set the correct L1 gas prices through
+        // inherents.
```
[the remark this answers is not in the document]
Comment (apoorvsadana) [reply crates/node/src/genesis_block.rs:66]: We need to manually enable it. I will open a PR soon to expose a enable/disable fees runtime call (not exposed on RPC). We're doing this to setup a shadow node for Kakarot as well.

File: crates/node/src/service.rs [hunk 2]
```diff
@@ -316,6 +324,43 @@ pub fn new_full(config: Configuration, sealing: SealingMode) -> Result<TaskManag
     );
 
     if role.is_authority() {
+        let l1_gas_price = Arc::new(Mutex::new(L1GasPrices::default()));
+
+        // initialize settlement workers
+        if let Some((layer_kind, config_path)) = settlement_config {
+            // TODO: make L1 message handling part of the SettlementProvider, support multiple layer options
+            if layer_kind == SettlementLayer::Ethereum {
```
[the remark this answers is not in the document]
Comment (apoorvsadana) [reply crates/node/src/service.rs:332]: it got removed by mistake here #1580

File: crates/client/rpc/src/runtime_api.rs [hunk 3]
```diff
@@ -104,13 +106,19 @@ where
...
+            .runtime_api()
+            .estimate_fee(block_hash, transactions, simulation_flags)
+            .map_err(|e: ApiError| {
                 error!("Request parameters error: {e}");
                 StarknetRpcApiError::InternalServerError
-            },
-        )???)
+            })???
+            .iter()
+            .map(|esimtate| esimtate.into())
```
[the remark this answers is not in the document]
Comment (apoorvsadana) [reply crates/client/rpc/src/runtime_api.rs]: fixed

File: crates/pallets/starknet/src/tests/starknet_inherent.rs [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+        // setting it first time works
+        assert_ok!(Starknet::set_starknet_inherent_data(
+            none_origin.clone(),
+            StarknetInherentData { sequencer_address: DEFAULT_SEQUENCER_ADDRESS, l1_gas_price: l1_gas_price.clone() }
+        ));
+        assert_eq!(Starknet::current_l1_gas_prices(), l1_gas_price);
+
+        let l1_gas_price_new = L1GasPrices { last_update_timestamp: 999, ..l1_gas_price };
+        // setting it second time causes it to panic
+        let _ = Starknet::set_starknet_inherent_data(
```
Comment (EvolveArt) [inline crates/pallets/starknet/src/tests/starknet_inherent.rs:90]: we dont assert the panic here ?

File: examples/messaging/eth-config.json [hunk 5]
```diff
@@ -0,0 +1,9 @@
+{
+  "provider": {
+    "rpc_endpoint": "https://eth.merkle.io",
```
Comment (EvolveArt) [inline examples/messaging/eth-config.json:3]: how will a node specify an rpc endpoint with their api key ? is that config hidden?

File: CHANGELOG.md [hunk 6]
```diff
@@ -2,6 +2,7 @@
 
 ## Next release
 
+- feat: actual estimate_fee added, brought back l1 messages
```
Comment (EvolveArt) [inline CHANGELOG.md:5]: also fixes simulation no?

**Comment by tdelabro** [comment]
@apoorvsadana will you finish this one?

**Comment by apoorvsadana** [comment]
I have done it from my side, need a review now @tdelabro

**Comment by tdelabro** [comment]
Oh sorry. Can you rebase?

**Comment by tdelabro** [comment]
Also I was thinking about the non answered review from @EvolveArt. Were those taken into account?

File: examples/messaging/eth-config.json (hunk 5 above)
Comment (apoorvsadana) [reply examples/messaging/eth-config.json:3]: yes, only the person running the node will have access to it. we can create an issue though to fetch values from the ENV if that's better?

File: crates/pallets/starknet/src/tests/starknet_inherent.rs (hunk 4 above)
Comment (apoorvsadana) [reply crates/pallets/starknet/src/tests/starknet_inherent.rs:90]: I added `#[should_panic]` at the start

File: CHANGELOG.md (hunk 6 above)
Comment (apoorvsadana) [reply CHANGELOG.md]: added

**Comment by apoorvsadana** [comment]
> Oh sorry. Can you rebase?

Done

> Also I was thinking about the non answered review from @EvolveArt. Were those taken into account?

My bad, missed the latest ones, replied to those

File: examples/messaging/eth-config.json (hunk 5 above)
Comment (EvolveArt) [reply examples/messaging/eth-config.json:3]: probably yes

**Review by EvolveArt — APPROVED** [review]

**Pull request merged**